crates/compiler/problem/src/can.rs

```diff
@@ -323,6 +322,7 @@ impl Problem {
             Problem::OverAppliedDbg { .. } => RuntimeError,
             Problem::DefsOnlyUsedInRecursion(_, _) => Warning,
             Problem::FileProblem { .. } => Fatal,
+            Problem::StmtAfterExpr(_) => Fatal,
```

Comment:
This only really comes up when purity inference is off (old platform) and I defaulted to `Fatal` because it was previously a parsing error. Since this is only temporary and maintains the status quo, I'd prefer to keep it like that to avoid potential issues. I'm happy to reconsider if you feel strongly about it, though.